rslib/src/sched.rs

```diff
@@ -132,15 +155,18 @@ mod test {
         assert_eq!(elap(crt, crt, offset, 4), 0);
         assert_eq!(elap(crt, crt - 86_400, offset, 4), 0);
 
-        // 2am the next day is still the same day
-        assert_eq!(elap(crt, crt + 24 * 3600, offset, 4), 0);
+        // day rolls over at 4am - crt is 2am
+        assert_eq!(elap(crt, crt + 2 * 3600, offset, 4), 1);
```

Comment:
I have been puzzling over creation time. Code appears to set it to 4am, as you say, yet in my collection, created April 24 2019, it is set to 3am. I was running Anki from the Linux Mint repo at the time. I have since uninstalled it so can't be sure what it was. This puzzle aside, if crt should always be 4am local time on the day of creation, it would make sense to have tests with this creation time. The implementation counts rollovers exclusive of the start time and inclusive of the end time so if creation time and rollover time are the same, the first day should be 0 days elapsed. This leaves the issue of what should be when creation time and rollover time are different - if the user changes rollover time, possibly long after creating the collection. I will add some tests with rollover time at 4am and see what happens.

For consistency with current implementation, if crt and rollover time are both 4am, after rollover time on the day of creation days elapsed should be 0 and after the rollover time on the day after creation, it should be 1, correct?

Reply:
The way Anki allowed the user to adjust the rollover point in the past was to modify the hour of the creation date - the code is in preferences.py. So perhaps you adjusted it at one point?

We can't assume the starting creation hour will match the current rollover hour though - the user may have adjusted the rollover in the past, or created their collection in a timezone other than the one they're in now.